packages/grafana-data/src/types/data.ts

```diff
@@ -87,6 +87,9 @@ export interface QueryResultMeta {
    */
   pathSeparator?: string;
 
+  /** A time shift diff provided with comparison query */
+  comparisonResponseDiff?: number;
```

Comment:
altNames: 

* timeCompareShiftMilliseconds (number)
* timeCompareShiftWindow (string), think it could be useful to have both. 
* timeCompareDataIsShifted (boolean to let time series panel know the timestamps have already been shifted) 

not sure if we should structure this as an object like this or keep it flat 

```
timeCompare: {
   milliseconds: number;
   window: string;
   isDataShifted: boolean; 
} 
```

Reply:
> timeCompareDataIsShifted (boolean to let time series panel know the timestamps have already been shifted)

any reason to have it pre-shifted? feels error-prone.

Reply:
> any reason to have it pre-shifted? feels error-prone.

A pretty huge benefit would be that all existing visualizations can handle it without any additional change / added complexity

Reply:
@leeoniya 

> any reason to have it pre-shifted? feels error-prone.

Why would it be more error-prone? 

Feels a lot less complex and error prone than https://github.com/grafana/grafana/pull/72715